packages/react/avatar/src/Avatar.test.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { axe } from 'jest-axe';
+import type { RenderResult } from '@testing-library/react';
+import { render, waitFor } from '@testing-library/react';
+import { getPartDataAttr } from '@radix-ui/utils';
+import * as Avatar from './Avatar';
+
+const ROOT_TEST_ID = 'avatar-root';
+const FALLBACK_TEXT = 'AB';
+const DELAY = 300;
+
+describe('given an Avatar with fallback and no image', () => {
+  let rendered: RenderResult;
+  let root: HTMLElement;
+  let fallback: HTMLElement;
+
+  beforeEach(() => {
+    rendered = render(
+      <Avatar.Root data-testid={ROOT_TEST_ID}>
+        <Avatar.Fallback>{FALLBACK_TEXT}</Avatar.Fallback>
+      </Avatar.Root>
+    );
+    root = rendered.getByTestId(ROOT_TEST_ID);
+    fallback = rendered.getByText(FALLBACK_TEXT);
+  });
+
+  it('should have no accessibility violations', async () => {
+    expect(await axe(rendered.container)).toHaveNoViolations();
+  });
+
+  it('should have a radix attribute on the root', () => {
+    const partDataAttr = getPartDataAttr('Avatar');
+    expect(root).toHaveAttribute(partDataAttr);
+  });
+
+  it('should have a radix attribute on the fallback', () => {
+    const partDataAttr = getPartDataAttr('AvatarFallback');
+    expect(fallback).toHaveAttribute(partDataAttr);
+  });
+});
+
+describe('given an Avatar with fallback and a working image', () => {
+  let rendered: RenderResult;
+
+  const orignalImage = window.Image;
+  beforeAll(() => {
+    jest.useFakeTimers();
+    (window.Image as any) = class MockImage {
+      onload: () => void = () => {};
+      src: string = '';
+      constructor() {
+        setTimeout(() => {
+          this.onload();
+        }, DELAY);
+        return this;
+      }
+    };
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+    window.Image = orignalImage;
+  });
+
+  beforeEach(() => {
+    rendered = render(
+      <Avatar.Root data-testid={ROOT_TEST_ID}>
+        <Avatar.Fallback>{FALLBACK_TEXT}</Avatar.Fallback>
+        <Avatar.Image data-testid="tst-123">{FALLBACK_TEXT}</Avatar.Image>
+      </Avatar.Root>
+    );
+  });
+
+  it('should render the fallback initially', () => {
+    const fallback = rendered.queryByText(FALLBACK_TEXT);
+    expect(fallback).toBeInTheDocument();
+  });
+
+  it('should not render the image initially', () => {
+    const image = rendered.queryByRole('img');
+    expect(image).not.toBeInTheDocument();
+  });
+
+  // TODO: MockImage isn't working as expected. Come back and figure out why.
+  // it('should render the image after it has loaded', async () => {
+  //   let image: HTMLElement | null = null;
+  //   await waitFor(() => {
+  //     jest.advanceTimersByTime(DELAY);
+  //   });
+  //   image = rendered.queryByTestId('tst-123');
+  //   expect(image).toBeInTheDocument();
+  // });
```

Comment:
I was debugging this for a bit and found out that the onload of the `useImageLoadingStatus` function isn't even getting called because in the test we are not providing "src" prop to Avatar.Image, and looking at the logic of `useImageLoadingStatus` it will just return from the useEffect if src isn't passed. 

https://github.com/radix-ui/primitives/blob/8663e7602f60a925fce34bfed3ff032c08ac2e06/packages/react/avatar/src/Avatar.tsx#L108-L111

Reply:
Great find, thank you for catching that!